relayer/src/tx/connection.rs

```diff
@@ -0,0 +1,121 @@
+use crate::chain::{Chain, CosmosSDKChain};
+use crate::config::ChainConfig;
+use crate::error::{Error, Kind};
+use ibc::ics24_host::identifier::{ClientId, ConnectionId};
+use ibc_proto::connection::{Counterparty, MsgConnectionOpenInit};
+use ibc_proto::tx::v1beta1::TxBody;
+use std::str::FromStr;
+use tendermint::account::Id;
+
+#[derive(Clone, Debug)]
+pub struct ConnectionOpenInitOptions {
+    pub src_client_id: ClientId,
+    pub dest_client_id: ClientId,
+    pub src_connection_id: ConnectionId,
+    pub dest_connection_id: ConnectionId,
+    pub src_chain_config: ChainConfig,
+    pub dest_chain_config: ChainConfig,
+}
+
+pub fn conn_init(opts: ConnectionOpenInitOptions) -> Result<(), Error> {
+    // Get the destination chain
+    let dest_chain = CosmosSDKChain::from_config(opts.clone().dest_chain_config)?;
+
+    let signer = Id::from_str(dest_chain.config().account_prefix.as_str()).map_err(|e| {
+        Kind::MessageTransaction("Connection Open Init: Bad Signer".into()).context(e)
+    })?;
+
+    let counterparty = Some(Counterparty {
+        client_id: opts.dest_client_id.to_string(),
+        connection_id: opts.dest_connection_id.to_string(),
+        prefix: None, // TODO Use a MerklePrefix
+    });
+
+    // MsgConnectionOpenInit protobuf message
+    let msg = MsgConnectionOpenInit {
+        client_id: opts.src_client_id.to_string(),
+        connection_id: opts.src_connection_id.to_string(),
+        counterparty,
+        signer: signer.as_bytes().to_vec(),
+    };
+
+    let mut proto_msgs: Vec<prost_types::Any> = Vec::new();
+    let mut buf = Vec::new();
+
+    // Have a loop if new_builder takes more messages
+    // for now just encode one message
+    prost::Message::encode(&msg, &mut buf).unwrap();
+
+    // Create a proto any message
+    let any_msg = prost_types::Any {
+        type_url: "/ibc.connection.MsgConnectionOpenInit".to_string(), // "type.googleapis.com/ibc.connection.MsgConnectionOpenInit".to_string(),
+        value: buf,
+    };
+
+    // Add proto message
+    proto_msgs.push(any_msg);
+
+    // Create TxBody
+    let body = TxBody {
+        messages: proto_msgs,
+        memo: "".to_string(),
+        timeout_height: 0,
+        extension_options: Vec::<prost_types::Any>::new(),
+        non_critical_extension_options: Vec::<prost_types::Any>::new(),
```

Comment:
Yes, this was moved to a Txbuilder logic (but could be part of the chain) that I will use when implementing the signing, this needs to be generic because that's how txs are sent to the chain and the signing can also be included in there.